src/GafferScene/SceneAlgo.cpp

```diff
@@ -662,9 +662,10 @@ SceneAlgo::History::Ptr SceneAlgo::history( const Gaffer::ValuePlug *scenePlugCh
 
 	CapturingMonitorPtr monitor = new CapturingMonitor;
 	{
-		ScenePlug::PathScope pathScope( Context::current(), path );
+		ScenePlug::PathScope pathScope( Context::current(), &path );
+		g_historyID++;
 		// Trick to bypass the hash cache and get a full upstream evaluation.
-		pathScope.set( historyIDContextName(), g_historyID++ );
+		pathScope.set( historyIDContextName(), &g_historyID );
```

Comment:
This is unsafe. `history()` can be called concurrently (and is, in the SceneViewInspector), so the value of `g_historyID` could change while we're using it. We need to take a copy to use locally. While we're at it, we should also use `std::atomic` for `g_historyID` so the increment is safe.

Reply:
Addressed in e15eea3.